complete/src/main/java/com/example/restservice/service/QueueService.java

```diff
@@ -24,10 +25,27 @@
   @Autowired private UserRepository userRepository;
   @Autowired private UserService userService; // TODO remove
 
+  private Random randomNumber = new Random();
   public CreateQueueResponse createQueue(CreateQueueRequest createQueueRequest) {
     try {
-      var queue = queueRepository.save(new Queue(createQueueRequest.getQueueName()));
-      return new CreateQueueResponse(queue.getQueueName(), queue.getQueueId());
+
+
+      Queue queueDao;
+      // if isPasswordProtected is true, generate a 4 digit pin.
+      if(Boolean.TRUE.equals(createQueueRequest.getIsPasswordProtected())) {
+        String queuePin=String.format("%04d", randomNumber.nextInt(10000));
+        queueDao=new Queue(createQueueRequest.getQueueName(),createQueueRequest.getIsPasswordProtected(),queuePin);
+        var queue = queueRepository.save(queueDao);
+        return new CreateQueueResponse(queue.getQueueName(), queue.getQueueId(),queue.getQueuePassword());
+      }
+      else{
+        queueDao =new Queue(createQueueRequest.getQueueName());
+        var queue = queueRepository.save(queueDao);
+        return new CreateQueueResponse(queue.getQueueName(), queue.getQueueId());
+      }
```

Comment:
```
String queuePassword = null;
if (createQueueRequest.getIsPasswordProtected()) {
  queuePassword = generateRandomPin();
}
var newQueue = new Queue(createQueueRequest.getQueueName(), queuePassword);
queueRepository.save(newQueue);
return new CreateQueueResponse(newQueue.getQueueName(), newQueue.getQueueId(), newQueue.getQueuePassword());
```

What say?

Reply:
That looks way better. I thought it is bad to send a null queuePassword value when it is not required or not set by business rules.. So used different constructors.

Reply:
Yes, can we somehow use Optional? If it's difficult to use Optional with Hibernate, lets no do that, but the getter can return an optional.